### ISSUE TITLE
Why is the distantly-supervised corrector trained with "all" labels?

### ISSUE THREAD

**Issue by S-Abdelnabi** [j6mes/acl2021-factual-error-correction#8]
Hi,
Thanks a lot for providing your code. 

As far as I understood from the paper, the corrector with distant supervision is trained as an autoencoder to reconstruct the claim [given the supporting evidence], such that it can be used later at test to make the claim consistent with the evidence. 

If that is true, then shouldn't the corrector be trained on "Supports" examples only? since the "refutes" are not consistent with the evidence, so they should not be reconstructed as they. Per the instructions to train the masker-corrector, it is trained on all labels. 

I hope you could please clarify this confusion.

**Comment by j6mes** [comment]
This is a selectable option with the parameters for the training script. Doing this doubles the number of available training data when reconstructing the sentences and empirically improves the performance of the final system.

For the NLI-masking based training this might invalidate the "correction" hypothesis we want to enforce by training. However the best training regimen was with random masks and the chance that the system will be trained with an imperfect instance still exists, but is a bit lower.

**Comment by S-Abdelnabi** [comment]
Got it, thanks a lot for the clarification!
